src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/DiagnosticsConfiguration.cs

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using System.Configuration;
+
+namespace System.Diagnostics
+{
+    internal enum InitState
+    {
+        NotInitialized,
+        Initializing,
+        Initialized
+    }
+
+    internal static class DiagnosticsConfiguration
+    {
+        private static volatile SystemDiagnosticsSection s_configSection;
+        private static volatile InitState s_initState = InitState.NotInitialized;
+
+#if !FEATURE_PAL // perfcounter
```

Comment:
I don't think you need the conditional compilation here.

Reply:
FEATURE_PAL can be assumed UNDEFINED ie you can remove the #if !FEATURE_PAL and retain everything within.

I believe this is a remnant from a very old port to Unix. PAL = Platform Abstraction Layer, and it's part of the old Unix build, which is no longer relevant.